src/billing_infos/swico/builder.rs

```diff
@@ -0,0 +1,169 @@
+use super::{
+    Arc, BillingInfos, Emitter, NaiveDate, StructuredSet, Swico, SwicoComponent, SwicoError,
+    TotalLenght, Version, DATE_FMT,
+};
+#[derive(Debug, Default, Clone)]
+pub struct S1Builder {
+    structured_set: StructuredSet,
+}
+impl S1Builder {
+    pub fn new() -> Self {
+        Self {
+            structured_set: StructuredSet::new(),
+        }
+    }
+    /// Add unstructured message into the billing informations
+    pub fn add_unstructured(&mut self, text: impl AsRef<str>) -> &mut Self {
+        self.structured_set
+            .insert(SwicoComponent::Unstructured, Arc::from(text.as_ref()));
+        self
+    }
+    /// Voucher/Invoice/Bill number
+    ///
+    /// The voucher date is the same as the date of the invoice
+    /// Tt is used as the reference date for the terms and conditions.
+    /// Together with the field /40/0:n, a maturity date of the invoice can be calculated
+    /// (payable within n days after the voucher date).
+    pub fn invoice_ref(&mut self, text: impl AsRef<str>) -> &mut Self {
+        self.structured_set
+            .insert(SwicoComponent::InvoiceRef, Arc::from(text.as_ref()));
+        self
+    }
+    /// Voucher/Invoice/Bill date
+    /// Accepted date format: "YYMMDD"
+    ///
+    /// # Caution:
+    ///
+    /// This can silently fail
+    /// Both [`"240101 (2024 January 1st)"`] and [`"010124 (2001 January 24th)"`] are valid dates
+    /// If you're unsure, use [`doc_date_naive()`] and pass a [`NaiveDate`] type
+    pub fn doc_date(&mut self, text: impl AsRef<str>) -> &mut Self {
+        self.structured_set
+            .insert(SwicoComponent::DocDate, Arc::from(text.as_ref()));
+        self
+    }
+    /// Voucher/Invoice/Bill date
+    pub fn doc_date_naive(&mut self, date: NaiveDate) -> &mut Self {
+        let text = format!("{}", date.format(DATE_FMT));
+        self.structured_set
+            .insert(SwicoComponent::DocDate, Arc::from(text.as_ref()));
+        self
+    }
+    /// Reference from the client/customer/debtor
+    ///
+    /// The customer reference is a reference sent by the customer
+    /// and is used to identify the bill
+    pub fn client_ref(&mut self, text: impl AsRef<str>) -> &mut Self {
+        self.structured_set
+            .insert(SwicoComponent::ClientRef, Arc::from(text.as_ref()));
+        self
+    }
+    /// TVA/MWST/VAT/IVA CH-UID From the creditor
+    ///
+    /// The VAT number is the same as the numerical UID of the service provider
+    /// (without the CHE prefix, separator and VAT suffix).
+    /// The VAT number can be used by the bill recipient to identify the bill issuer unambiguously.
+    /// All bill issuers who have a UID should enter it here,
+    /// even if the other VAT fields are omitted.
+    /// For a bill with more than one VAT number, the first should be entered
+    pub fn vat_num(&mut self, text: impl AsRef<str>) -> &mut Self {
```

Comment:
Can we call those `add_*` or `set_*`? Usually in rust without a prefix it's a getter and with it's a setter.